contracts/test-utils/src/assertions.ts

```diff
@@ -159,10 +187,76 @@ export async function expectTransactionFailedWithoutReasonAsync(p: sendTransacti
  * @returns a new Promise which will reject if the conditions are not met and
  * otherwise resolve with no value.
  */
-export async function expectContractCallFailedAsync<T>(p: Promise<T>, reason: RevertReason): Promise<void> {
+export async function expectContractCallFailedAsync<T>(
+    p: Promise<T>,
+    reason: RevertReason | RichRevertReason,
+): Promise<void> {
+    if (reason instanceof RichRevertReason) {
+        return _expectContractCallRichRevertAsync<T>(p, reason);
+    }
     return expect(p).to.be.rejectedWith(reason);
 }
 
+/**
+ * A hacky way of expecting a rich revert reason to resolve.
+ * This is because geth does not signal when a revert happens during an
+ * eth_call so we're left to guess whether the returned data is an encoded
+ * rich revert reason.
+ * @param p A Promise resulting from a contract call.
+ * @param reason a rich revert reason
+ * @returns a promise which will reject `p` does not resolve to a rich revert reason.
+ */
+async function _expectContractCallRichRevertAsync<T>(p: Promise<T>, reason: RichRevertReason): Promise<void> {
+    let decoded: RichRevertReason | undefined;
```

Comment:
We have a variable naming convention that if a variable might hold some value or `undefined`, we suffix it's name with `IfExists`, i.e., `decodedValueIfExists`. We also generally prefer verbose, readable variable names over concise, potentially confounding names. Reading code is hard enough as it is 🍅